File: solutions/1668. Maximum Repeating Substring/1668.py
def max_repeating(sequence: str, word: str) -> int:
    repeats = 0
    words = word
    while words in sequence:
        repeats += 1
        words += word
    return repeats


print(max_repeating('aaabaaaabaaabaaaabaaaabaaaabaaaaba', 'aaaba'))

File: solutions/1668. Maximum Repeating Substring/1668.go
package main

import "strings"

func maxRepeating(sequence string, word string) int {
	var maxRepeats int
	for k := 1; k <= len(sequence)/len(word); k++ {
		if strings.Contains(sequence, strings.Repeat(word, k)) {
			maxRepeats = k
		} else {
			break
		}
	}
	return maxRepeats
}
func main() {
	sequence := "aaabaaaabaaabaaaabaaaabaaaabaaaaba"
	println(maxRepeating(sequence, "aaba"))

}
